### PR TITLE
refine avro serde type to SpecificAvroSerde and GernericAvroSerde

### PR DESCRIPTION
**Pull request by humkum**
1. provide specific avro serde & generic avro serde
2. change some  annotation message.

### CODE REVIEW

**Comment by MatrixHB** [comment]
What is the relation between SpecificAvroSerde, GernericAvroSerde and AvroSerde？Up-down dependency, or parallel ?

If SpecificAvroSerde and GernericAvroSerde are exposed to users and they are dependent on AvroSerde, I think AvroSerde should be moved to common module, to avoid user confusion. 
If they are parallel, it is better to merge the Serde package and the Serializer package. Or I will feel a bit confused about which to choose.

@humkum @ferrirW

**Comment by MatrixHB** [comment]
Is it better to make a "avro" package under the "serde" path, leave expansion room for JSON serde and Protobuf serde ?

File: example/src/main/java/org/apache/rocketmq/schema/registry/example/serde/SpecificAvroSerdeDemo.java [hunk 1]
```diff
@@ -19,26 +19,28 @@
...
+import org.apache.rocketmq.schema.registry.client.serde.SpecificAvroSerde;
 
-public class AvroSerdeDemo {
+import java.io.IOException;
+
+public class SpecificAvroSerdeDemo {
 
     public static void main(String[] args) {
 
         String baseUrl = "http://localhost:8080/schema-registry/v1";
```
Comment (ferrirW) [inline example/src/main/java/org/apache/rocketmq/schema/registry/example/serde/SpecificAvroSerdeDemo.java:30]: Instead, mock a SchemaRegistry in a unit test.

**Review by ferrirW — APPROVED** [review]

**Comment by humkum** [comment]
> What is the relation between SpecificAvroSerde, GernericAvroSerde and AvroSerde？Up-down dependency, or parallel ?
> 
> If SpecificAvroSerde and GernericAvroSerde are exposed to users and they are dependent on AvroSerde, I think AvroSerde should be moved to common module, to avoid user confusion. If they are parallel, it is better to merge the Serde package and the Serializer package. Or I will feel a bit confused about which to choose.
> 
> @humkum @ferrirW

I can't agree more with you, their relationship is an Up-down dependency, and I agree to move AvroSerde to common package

**Comment by MatrixHB** [comment]
> > What is the relation between SpecificAvroSerde, GernericAvroSerde and AvroSerde？Up-down dependency, or parallel ?
> > If SpecificAvroSerde and GernericAvroSerde are exposed to users and they are dependent on AvroSerde, I think AvroSerde should be moved to common module, to avoid user confusion. If they are parallel, it is better to merge the Serde package and the Serializer package. Or I will feel a bit confused about which to choose.
> > @humkum @ferrirW
> 
> I can't agree more with you, their relationship is an Up-down dependency, and I agree to move AvroSerde to common package

I‘m sorry. I try to move to common module but it's more unreasonable because AvroSerde is dependent on SchemaRegistryClient, so let's just keep this way.
I guess you have learned from kafka-schema-registry, so I agree with your current design.

--------------------------
_Is it better to make a "avro" package under the "serde" path, leave expansion room for JSON serde and Protobuf serde ?_
BTW, you can take this into consideration.

**Comment by humkum** [comment]
> > > What is the relation between SpecificAvroSerde, GernericAvroSerde and AvroSerde？Up-down dependency, or parallel ?
> > > If SpecificAvroSerde and GernericAvroSerde are exposed to users and they are dependent on AvroSerde, I think AvroSerde should be moved to common module, to avoid user confusion. If they are parallel, it is better to merge the Serde package and the Serializer package. Or I will feel a bit confused about which to choose.
> > > @humkum @ferrirW
> > 
> > 
> > I can't agree more with you, their relationship is an Up-down dependency, and I agree to move AvroSerde to common package
> 
> I‘m sorry. I try to move to common module but it's more unreasonable because AvroSerde is dependent on SchemaRegistryClient, so let's just keep this way. I guess you have learned kafka-schema-registry, so I agree with your current design.
> 
> _Is it better to make a "avro" package under the "serde" path, leave expansion room for JSON serde and Protobuf serde ?_ BTW, you can take this into consideration.

Okay , it's a future plan.

**Review by MatrixHB — APPROVED** [review]

**Review by duhenglucky — APPROVED** [review]

File: example/src/main/java/org/apache/rocketmq/schema/registry/example/serde/GenericAvroSerdeDemo.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            Map<String, Object> configs = new HashMap<>();
+            configs.put(AvroDeserializerConfig.USE_GENERIC_DATUM_READER, true);
+            serde.configure(configs);
+
+            //serialize
+            byte[] bytes = serde.serializer().serialize("TopicTest", record);
+
+            //deserialize
+            GenericRecord record1 = serde.deserializer().deserialize("TopicTest", bytes);
+            System.out.println("the origin object after ser/de is " + record1);
```
Comment (ShannonDing) [inline example/src/main/java/org/apache/rocketmq/schema/registry/example/serde/GenericAvroSerdeDemo.java:55]: print origin record to show if it has different from result record.

**Pull request merged**